src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TreeDiffer.java

```diff
@@ -197,13 +201,28 @@ public void visitIdent(JCIdent tree) {
                 return;
             }
         }
-        result = tree.sym == that.sym;
+        result = scan(symbol, otherSymbol);
+    }
+
+    private boolean scan(Symbol symbol, Symbol otherSymbol) {
```

Comment:
Consider renaming this to `scanSymbol`, to make it clear it doesn't apply to trees.